src/adapter/db/wishlist.gateway.ts

```diff
@@ -0,0 +1,68 @@
+import { EntityManager } from '@mikro-orm/core';
+import { EntityRepository } from '@mikro-orm/mysql';
+import { InjectRepository } from '@mikro-orm/nestjs';
+import { Inject, Injectable } from '@nestjs/common';
+import { WishlistNotFoundException } from 'src/application/common/error/exception/wishlist.exception';
+import { AddWishlistCommand } from 'src/application/port/in/wishlist/WishlistUseCase';
+import { WishlistDbCommandPort } from 'src/application/port/out/WishlistDbCommandPort';
+import { WishlistDbQueryPort } from 'src/application/port/out/WishlistDbQueryPort';
+import { Wishlist } from 'src/domain/wishlist';
+
+import { WishlistDbEntity } from './wishlist.entity';
+import { mapToWishlist } from './wishlist.mapper';
+
+@Injectable()
+export class WishlistGateway implements WishlistDbCommandPort, WishlistDbQueryPort {
+  constructor(
+    @InjectRepository(WishlistDbEntity)
+    private readonly wishlistRepository: EntityRepository<WishlistDbEntity>,
+    @Inject(EntityManager)
+    private readonly em: EntityManager,
+  ) {}
+
+  async getWishlistById(wishlistId: number): Promise<Wishlist> {
+    const wishlist = await this.wishlistRepository.findOne({ id: wishlistId });
+
+    if (!wishlist) {
+      throw new WishlistNotFoundException();
+    }
+
+    return mapToWishlist(wishlist);
+  }
+
+  async getAllByUserId(userId: number): Promise<Wishlist[]> {
+    const entities = await this.wishlistRepository.find(
+      { userId },
+      { orderBy: { createdAt: 'desc' } },
+    );
+    return entities.map(mapToWishlist);
+  }
```

Comment:
data layer 관점에서 보면 특정 유저의 wishlist는 all(모든 데이터)가 아닙니다!
```suggestion
  async getUserWishlistByUserId(userId: number): Promise<Wishlist[]> {
    const entities = await this.wishlistRepository.find(
      { userId },
      { orderBy: { createdAt: 'desc' } },
    );
    return entities.map(mapToWishlist);
  }
```

Reply:
수정했습니다!